### PR TITLE
Use scratch as the base image for scheduler containers

### PR DESCRIPTION
**Pull request by Chrisbattarbee**
Reduce the size and attack surface of scheduler containers by using a base image of scratch

Tested locally on a minikube cluster

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
 
+FROM scratch
+
+ENV SERVICE_HOME=/opt/palantir/services/spark-scheduler
+
+# Copy the spark-scheduler user from the alpine stage
+COPY --from=0 /etc/passwd /etc/passwd
+
+# Copy the service home from the alpine stage
+COPY --from=0 $SERVICE_HOME $SERVICE_HOME
```
Comment (VVKot) [inline docker/Dockerfile:20]: nit: can't we just `ENV SERVICE_HOME=/opt/palantir/services/spark-scheduler` in this stage only? Same goes for SSCW image

**Review by VVKot — APPROVED** [review]
Looks good otherwise

File: docker/Dockerfile (hunk 1 above)
Comment (VVKot) [reply docker/Dockerfile:20]: Ah nvm, we have to copy that because only `alpine` knows how to `mkdir`.

**Review by rbotarleanu — APPROVED** [review]
LGTM

**Pull request merged**